components/client/typescript/src/server.ts

```diff
@@ -118,16 +122,61 @@ export async function buildServer(
     }
   }
 
-  async function registerPredicates(this: FastifyInstance) {
+  async function isPredicateActive(predicate: ServerPredicate): Promise<boolean | undefined> {
+    try {
+      const result = await request(`${chainhookOpts.base_url}/v1/chainhooks/${predicate.uuid}`, {
+        method: 'GET',
+        headers: { accept: 'application/json' },
+        throwOnError: true,
+      });
+      const response = (await result.body.json()) as SerializedPredicateResponse;
+      if (response.status == 404) return undefined;
+      if (
+        response.result.enabled == false ||
+        response.result.status.type == 'interrupted' ||
+        response.result.status.type == 'unconfirmed_expiration' ||
+        response.result.status.type == 'confirmed_expiration'
+      ) {
+        return false;
+      }
+      return true;
+    } catch (error) {
+      logger.error(
+        error,
+        `ChainhookEventObserver unable to check if predicate ${predicate.uuid} is active`
+      );
+      return false;
+    }
+  }
+
+  async function registerAllPredicates(this: FastifyInstance) {
+    logger.info(predicates, `ChainhookEventObserver connected to ${chainhookOpts.base_url}`);
     if (predicates.length === 0) {
       logger.info(`ChainhookEventObserver does not have predicates to register`);
       return;
     }
     const nodeType = serverOpts.node_type ?? 'chainhook';
     const path = nodeType === 'chainhook' ? `/v1/chainhooks` : `/v1/observers`;
     const registerUrl = `${chainhookOpts.base_url}${path}`;
-    logger.info(predicates, `ChainhookEventObserver registering predicates at ${registerUrl}`);
     for (const predicate of predicates) {
+      if (nodeType === 'chainhook') {
+        switch (await isPredicateActive(predicate)) {
+          case undefined:
+            // Predicate doesn't exist.
+            break;
+          case true:
+            logger.info(
+              `ChainhookEventObserver predicate ${predicate.uuid} is already active, skipping registration`
+            );
+            continue;
+          case false:
+            logger.info(
+              `ChainhookEventObserver predicate ${predicate.uuid} was being used but is now inactive, updating`
```

Comment:
This log information suggests that an inactive predicate will be updated somehow, rather than deleted. Perhaps changing the information to `... was being used but is now inactive, removing it` would be more accurate.